.github/workflows/main.yml

```diff
@@ -631,22 +631,40 @@ jobs:
       env:
         RUST_BACKTRACE: 1
 
-    # Test debug (DWARF) related functionality.
+    # NB: the test job here is explicitly lacking in cancellation of this run if
+    # something goes wrong. These take the longest anyway and otherwise if
+    # Windows fails GitHub Actions will confusingly mark the failed Windows job
+    # as cancelled instead of failed.
+
+  # Test debug (DWARF) related functionality.
+  test_debug_dwarf:
+    needs: determine
+    name: Test DWARF debugging
+    runs-on: 'ubuntu-latest'
+    strategy:
+      fail-fast: ${{ github.event_name != 'pull_request' }}
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        submodules: true
+    - uses: ./.github/actions/install-rust
+    - run: rustup target add wasm32-wasi wasm32-unknown-unknown
     - run: |
         sudo apt-get update && sudo apt-get install -y gdb lldb-15 llvm
         # woraround for https://bugs.launchpad.net/ubuntu/+source/llvm-defaults/+bug/1972855
         sudo mkdir -p /usr/lib/local/lib/python3.10/dist-packages/lldb
         sudo ln -s /usr/lib/llvm-15/lib/python3.10/dist-packages/lldb/* /usr/lib/python3/dist-packages/lldb/
         cargo test test_debug_dwarf -- --ignored --test-threads 1
-      if: matrix.os == 'ubuntu-latest' && matrix.target == ''&& needs.determine.outputs.run-full
+      if: needs.determine.outputs.run-full
```

Comment:
This `if` can now move up into the header under `needs` for example to skip the whole job for PRs for example